MekHQ/src/mekhq/campaign/mission/AtBContract.java

```diff
@@ -238,41 +243,32 @@ public static Camouflage pickRandomCamouflage(int currentYear, String factionCod
         final String ROOT_DIRECTORY = "data/images/camo/";
 
         String camouflageDirectory = getCamouflageDirectory(currentYear, factionCode);
-
-        // Use Java File to represent directories
-        File workingDirectory = new File(ROOT_DIRECTORY + camouflageDirectory + '/');
-
-        // List subdirectories and loose files in working directory
-        File[] folders = workingDirectory.listFiles(File::isDirectory);
-        File[] looseFiles = workingDirectory.listFiles();
-
+      
         // Gather all files
-        List<File> allFiles = new ArrayList<>();
-        if (looseFiles != null) {
-            Collections.addAll(allFiles, looseFiles);
-        }
-        if (folders != null) {
-            for (File folder : folders) {
-                File[] folderFiles = folder.listFiles();
-                if (folderFiles != null) {
-                    Collections.addAll(allFiles, folderFiles);
-                }
-            }
-        }
+        List<Path> allPaths = null;
 
+        try {
+            allPaths = Files.find(Paths.get(ROOT_DIRECTORY + camouflageDirectory + '/'), Integer.MAX_VALUE,
+                    (path, bfa) -> {return bfa.isRegularFile();})
+                .collect(Collectors.toList());
+        } catch (IOException e) {
+            logger.error("Error getting list of camouflages", e);
+        }
+        
         // Select a random file to set camouflage, if there are files available
-        if (!allFiles.isEmpty()) {
-            File randomFile = allFiles.get(new Random().nextInt(allFiles.size()));
+        if ((null != allPaths) && (!allPaths.isEmpty())) {
+            Path randomPath = allPaths.get(new Random().nextInt(allPaths.size()));
 
-            String fileName = randomFile.getName();
-            String fileCategory = randomFile.getParent().replaceAll("\\\\", "/");
+            String fileName = randomPath.getFileName().toString();
+            String fileCategory = randomPath.getParent().toString()
+                .replaceAll("\\\\", "/"); // Is this necessary on windows machines?
```

Comment:
Yes, to answer your question. Basically, windows gets angry if you're using mixed slashes in a directory string. So we convert to use all forward slashes, which it is happy to accept - because all the slashes are in the same direction. Windows is specifically the reason we do this.

There is almost certainly a better way to handle it.

Reply:
If I was still on windows I'd throw a log on there and confirm if the file functions are actually giving us backslashes because MOST development systems will just give forward slashes on all systems, but I don't know about java... :|

Reply:
When we initially launched this we ran into issues when using mixed slashes, because `getParent` returns `parent/child` on good OSs and `parent\child` on windows. So you end up in a situation where you're seeing `parent\child/myLittleCamo.png` which was confusing the heck out of windows.

Reply:
Or something to that effect